cmd/arc/services/metamorph.go

```diff
@@ -201,19 +199,10 @@ func StartMetamorph(logger *slog.Logger, arcConfig *config.ArcConfig, cacheStore
 			continue
 		}
 
-		zmq := metamorph.NewZMQ(zmqURL, statusMessageCh, logger)
+		zmq := metamorph.NewZMQ(context.Background(), zmqURL, statusMessageCh, nil, logger)
```

Comment:
IMO we should create zmq handler here and pass it
```suggestion
zmqHandler := NewZMQHandler(ctx, zmqURL, logger)
zmq := metamorph.NewZMQ(context.Background(), zmqURL, statusMessageCh, zmqHandler, logger)
```

Reply:
Done